controllers/drplacementcontrol_controller.go

```diff
@@ -1264,9 +1264,16 @@ func (r *DRPlacementControlReconciler) finalizeDRPC(ctx context.Context, drpc *r
 		return fmt.Errorf("VRG adoption in progress")
 	}
 
-	// delete manifestworks (VRGs)
+	// delete manifestworks (VRGs and NSs)
 	for _, drClusterName := range rmnutil.DRPolicyClusterNames(drPolicy) {
-		err := mwu.DeleteManifestWorksForCluster(drClusterName)
+		err := mwu.DeleteVRGManifestWorksForCluster(drClusterName)
```

Comment:
We need to ensure that the NS MW has the delete option of orphan here, such that upgrades that lead to deletion of the NS guarantees that we will not delete the NS on the managed cluster.

So at the finalize part, we would need to ensure that the NS MW has the delete option, and if not call CreateOrUpdateNamespaceManifest such that it is updated with the delete option, and then delete the MW.